src/sql/workbench/contrib/assessment/common/asmtActions.ts

```diff
@@ -198,8 +211,8 @@ export class AsmtExportAsScriptAction extends Action {
 
 	public async run(context: IAsmtActionInfo): Promise<boolean> {
 		this._telemetryService.sendActionEvent(TelemetryView.SqlAssessment, AsmtExportAsScriptAction.ID);
-		if (context && context.component && context.component.resultItems) {
-			await this._assessmentService.generateAssessmentScript(context.ownerUri, context.component.resultItems);
+		if (context && context.component && context.component.recentResult?.result.items) {
+			await this._assessmentService.generateAssessmentScript(context.ownerUri, context.component.recentResult?.result.items);
```

Comment:
Since you use this twice I'd just pull it into its own var outside the if statement to keep things shorter.